### DISCUSSION TITLE
Relate internal mol coordinates with MolToImage() atom positions in pixels

### DISCUSSION THREAD

**Discussion by MichailDanikas** [rdkit/rdkit#6128] [Q&A] [answered]
Hi,

I initialized my molecule using smiles and then I saved it as an image using MolToImage() method. Let's say that my image is size = (300, 300). Is there a way to get the position of each atom in the depicted molecule in pixels?
I know that using the method `GetConformer().GetAtomPosition(idx)` can give me the internal coordinates of the atom with index `idx`.

My ultimate goal is to plot the image using Matplotlib (`plt.imshow(img)`) and with the onclick event I can click on any atom of the molecule and print their corresponding index for later usage.

I also want to avoid using MolDraw2DCairo() and use only MolToImage() method.

Thank you in advance.

**Comment by greglandrum** [comment, marked as answer]
> 
> I also want to avoid using MolDraw2DCairo() and use only MolToImage() method.

Before I answer the rest of the question: why do you want to do this? For advanced usage (like what you're trying here), you have a lot more control when you use either `MolDraw2DCairo` (for PNG images) or `MolDraw2DSVG` (for SVGs).

**Reply by MichailDanikas** [reply]
I have a large database with molecules saved as images with their corresponding SMILES. These molecules are depicted as being protonated/ deprotonated but their SMILES are not correct. So I want to have a 2x1 plot where on 1x1 the molecule from the database is depicted and on 2x1 I generate my neutral one from a list. The idea is to use matplotlib as an interface to click on my atom's position in which the atom on the image from the database is protonated/ deprotonated and then generate the correct SMILES using `GetAtomWithIdx(atom_id).SetFormalCharge(charge)`. 

It seems that the molecules from the database were saved with `MolToImage()` and it would be convenient for me to use the same method in order to make the comparison easier. Using `MolDraw2DCairo()` the depiction of the molecule is different than the former method.

However, I already started working using `MolDraw2DCairo()` but I was just curious if this is possible also with `MolToImage()`.
![rdkit_qa](https://user-images.githubusercontent.com/101356745/221517561-d5ca88e8-1cae-4446-9816-25a83c77ed67.png)

**Reply by greglandrum** [reply]
The best way to do this that I'm aware of is to use MolDraw2DSVG and the TagAtoms() function, which will put a transparent circle around each atom that can be used for selection. There's some sample code out there to do this. 
Here's one example:
https://www.cresset-group.com/about/news/review-8th-rdkit-ugm/
and here's another:
https://www.cheminformania.com/learn-how-to-make-a-jupyter-notebook-widget-for-annotation-of-atom-properties/

**Reply by greglandrum** [reply]
If, for whatever reason, you don't want to go down that route, you can ask a MolDraw2DSVG or MolDraw2DCairo object to map molecule coordinates to drawing coordinates with the method `GetDrawCoords()`:
```
d2d = Draw.MolDraw2DSVG(250,300)
d2d.DrawMolecule(m)
d2d.FinishDrawing()

from rdkit import Geometry
conf = m.GetConformer()
drawCoords = [d2d.GetDrawCoords(Geometry.Point2D(conf.GetAtomPosition(i))) for i in range(m.GetNumAtoms())]
```
That demo is with a MolDraw2DSVG object, but it works exactly the same with a MolDraw2DCairo object.

**Reply by MichailDanikas** [reply]
Thank you so much for your helpful input! I wasn't aware of the examples within the links and they helped a lot.
